Remotion/Validation/UnitTests/RuleBuilders/AddingObjectValidationRuleBuilderTest.cs

```diff
@@ -14,61 +14,148 @@
     {
+      _addingObjectValidationRuleCollectorMock.Setup(mock => mock.SetRemovable()).Verifiable();
+
+      _addingObjectValidationBuilder.CanBeRemoved();
+
+      _addingObjectValidationRuleCollectorMock.Verify();
     }
 
     [Test]
-    [Ignore("TODO RM-5906")]
     public void AddMetaValidationRule ()
     {
+      var metaValidationRuleStub = new Mock<IObjectMetaValidationRule>();
+
+      _objectMetaValidationRuleCollectorMock.Setup(mock => mock.RegisterMetaValidationRule(metaValidationRuleStub.Object)).Verifiable();
+
+      _addingObjectValidationBuilder.AddMetaValidationRule(metaValidationRuleStub.Object);
+
+      _objectMetaValidationRuleCollectorMock.Verify();
     }
 
     [Test]
-    [Ignore("TODO RM-5906")]
     public void AddMetaValidationRule_FuncOverload ()
     {
+      var fakeValidationResult = MetaValidationRuleValidationResult.CreateValidResult();
+
+      _objectMetaValidationRuleCollectorMock
+          .Setup(
+              mock => mock.RegisterMetaValidationRule(
+                  It.Is<IObjectMetaValidationRule>(
+                      rule => rule.GetType() == typeof(DelegateObjectMetaValidationRule<IObjectValidator>)
+                              && rule.Validate(new IObjectValidator[0]).First() == fakeValidationResult)))
+          .Verifiable();
+
+      _addingObjectValidationBuilder.AddMetaValidationRule(v => fakeValidationResult);
+
+      _objectMetaValidationRuleCollectorMock.Verify();
     }
 
     [Test]
-    [Ignore("TODO RM-5906")]
     public void AddMetaValidationRule_ExpressionOverload_IsValidIsTrue ()
     {
+      _objectMetaValidationRuleCollectorMock
+          .Setup(
+              mock => mock.RegisterMetaValidationRule(
+                  It.Is<IObjectMetaValidationRule>(
+                      rule => rule.GetType() == typeof(DelegateObjectMetaValidationRule<IObjectValidator>)
+                              && rule.Validate(new IObjectValidator[0]).First().IsValid)))
+          .Verifiable();
+
+      _addingObjectValidationBuilder.AddMetaValidationRule<IObjectValidator>(v => true);
+
+      _objectMetaValidationRuleCollectorMock.Verify();
     }
 
     [Test]
-    [Ignore("TODO RM-5906")]
     public void AddMetaValidationRule_ExpressionOverload_IsValidIsFalse ()
     {
+      _objectMetaValidationRuleCollectorMock
+          .Setup(
+              mock => mock.RegisterMetaValidationRule(
+                  It.Is<IObjectMetaValidationRule>(
+                      rule => rule.GetType() == typeof(DelegateObjectMetaValidationRule<IObjectValidator>)
+                              && !rule.Validate(new IObjectValidator[0]).First().IsValid
+                              && rule.Validate(new IObjectValidator[0]).First().Message
+                              == "Meta validation rule 'v => False' failed for validator 'Remotion.Validation.Validators.IObjectValidator' "
+                              + "on type 'Remotion.Validation.UnitTests.TestDomain.Customer'.")))
+          .Verifiable();
```

Comment:
I would prefer we captur the rule via a callback and do the assertion a the end of the test. that way, such a complex assertion logic can provide helpful diagnostic. now, if any of the boolean conditions fails, the test is not matched and we have a really hard time figuring out what went wrong. This applies to the other tests, too.